# YOLO Model Comparison for Dish Detection

This notebook compares three YOLO object detection models to determine the best architecture for detecting dishes in food images. We evaluate:

1. **YOLOv8-n** (nano) - ~3.2M parameters, fastest inference
2. **YOLOv8-s** (small) - ~9.8M parameters, balanced approach
3. **YOLO11-n** (nano) - Latest architecture, improved accuracy

All models are trained on the same dataset with identical hyperparameters for a fair comparison.

## Configuration

Set up training parameters and paths:
- **Image size**: 768px for higher resolution detection
- **Epochs**: 80 training epochs
- **Batch size**: 16 images per batch
- **Device**: GPU (CUDA device 0)

We also reference a pre-trained YOLOv8-n model from earlier training runs for comparison.

In [9]:
from ultralytics import YOLO
import os
import pandas as pd

DATA_YAML = "data_yolo_dish/data.yaml"
IMG_SIZE = 768
EPOCHS = 80
BATCH_SIZE = 16
DEVICE = 0

BEST_YOLOV8N = r"runs/detect/train_dishdet_multi3/weights/best.pt"


## Train YOLOv8-s (Small Variant)

Train YOLOv8-s as a baseline comparison. This model has ~9.8M parameters (3x larger than nano) and typically offers better accuracy at the cost of slower inference speed.

In [10]:
print("=" * 70)
print("Training YOLOv8-s for dish detection")
print("=" * 70)

model_y8s = YOLO("yolov8s.pt")

result_y8s = model_y8s.train(
    data=DATA_YAML,
    epochs=EPOCHS,
    imgsz=IMG_SIZE,
    batch=BATCH_SIZE,
    device=DEVICE,
    name="compare_yolov8s_dishdet",
    project="runs/detect",
    cache="ram",
    seed=0,
    deterministic=True,
)

BEST_YOLOV8S = os.path.join(
    "runs", "detect", "compare_yolov8s_dishdet", "weights", "best.pt"
)
print(f"\n[YOLOv8-s] Best weights: {BEST_YOLOV8S}")

print("\nValidating best YOLOv8-s weights on validation set")
y8s_best_model = YOLO(BEST_YOLOV8S)
metrics_y8s = y8s_best_model.val(
    data=DATA_YAML,
    split="val",
    imgsz=IMG_SIZE,
    batch=BATCH_SIZE,
)

print(
    f"[YOLOv8-s] P={metrics_y8s.box.mp:.3f}, "
    f"R={metrics_y8s.box.mr:.3f}, "
    f"mAP50={metrics_y8s.box.map50:.3f}, "
    f"mAP50-95={metrics_y8s.box.map:.3f}"
)


Training YOLOv8-s for dish detection
New https://pypi.org/project/ultralytics/8.3.233 available  Update with 'pip install -U ultralytics'
Ultralytics 8.3.5  Python-3.11.9 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)
engine\trainer: task=detect, mode=train, model=yolov8s.pt, data=data_yolo_dish/data.yaml, epochs=80, time=None, patience=100, batch=16, imgsz=768, save=True, save_period=-1, cache=ram, device=0, workers=8, project=runs/detect, name=compare_yolov8s_dishdet, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=

AMP: checks passed 


train: Scanning C:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\data_yolo_dish\train\labels.cache... 903 images, 0 backgrounds, 0 corrupt: 100%|██████████| 903/903 [00:00<?, ?it/s]

WARNING  Box and segment counts should be equal, but got len(segments) = 83, len(boxes) = 5212. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.


WARNING  cache='ram' may produce non-deterministic training results. Consider cache='disk' as a deterministic alternative if your disk space allows.


train: Caching images (1.0GB RAM): 100%|██████████| 903/903 [00:01<00:00, 607.46it/s]
val: Scanning C:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\data_yolo_dish\valid\labels.cache... 181 images, 0 backgrounds, 0 corrupt: 100%|██████████| 181/181 [00:00<?, ?it/s]

WARNING  Box and segment counts should be equal, but got len(segments) = 28, len(boxes) = 1004. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.
WARNING  cache='ram' may produce non-deterministic training results. Consider cache='disk' as a deterministic alternative if your disk space allows.



val: Caching images (0.2GB RAM): 100%|██████████| 181/181 [00:00<00:00, 1381.70it/s]


Plotting labels to runs\detect\compare_yolov8s_dishdet\labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 63 weight(decay=0.0), 70 weight(decay=0.0005), 69 bias(decay=0.0)
TensorBoard: model graph visualization added 
Image sizes 768 train, 768 val
Using 8 dataloader workers
Logging results to runs\detect\compare_yolov8s_dishdet
Starting training for 80 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/80      5.88G     0.5946      1.259      1.034        100        768: 100%|██████████| 57/57 [00:20<00:00,  2.80it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.26it/s]

                   all        181       1004      0.965        0.3      0.851      0.689



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       2/80      5.88G     0.5704     0.7284     0.9947         71        768: 100%|██████████| 57/57 [00:18<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.25it/s]

                   all        181       1004      0.831      0.797       0.85      0.686



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       3/80      5.89G     0.5943     0.6622      1.003        102        768: 100%|██████████| 57/57 [00:18<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.98it/s]

                   all        181       1004      0.732      0.766      0.803       0.61



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       4/80      5.93G     0.5722     0.6126     0.9923         60        768: 100%|██████████| 57/57 [00:17<00:00,  3.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.78it/s]

                   all        181       1004      0.756      0.769      0.776      0.604



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       5/80      5.87G     0.5555     0.5788     0.9817         57        768: 100%|██████████| 57/57 [00:17<00:00,  3.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.80it/s]

                   all        181       1004      0.804      0.844      0.834      0.685



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       6/80      5.88G     0.5366     0.5425     0.9721         65        768: 100%|██████████| 57/57 [00:17<00:00,  3.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.75it/s]

                   all        181       1004      0.844      0.842      0.877      0.727



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       7/80      5.85G     0.5253     0.5033     0.9656         84        768: 100%|██████████| 57/57 [00:18<00:00,  3.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.57it/s]

                   all        181       1004       0.83      0.849      0.887      0.757



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       8/80       5.9G     0.5255     0.4921     0.9565         78        768: 100%|██████████| 57/57 [00:18<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.72it/s]

                   all        181       1004      0.786      0.818      0.851      0.707



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       9/80      5.89G     0.5107     0.4731     0.9591         95        768: 100%|██████████| 57/57 [00:18<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.75it/s]

                   all        181       1004      0.811      0.851      0.882      0.759



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      10/80      5.92G     0.4893     0.4542     0.9446        125        768: 100%|██████████| 57/57 [00:18<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.85it/s]

                   all        181       1004      0.866      0.854      0.891      0.771



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      11/80      5.89G     0.4941     0.4434     0.9528         92        768: 100%|██████████| 57/57 [00:18<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.56it/s]

                   all        181       1004      0.811      0.788      0.858      0.736



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      12/80      5.84G     0.4832     0.4168     0.9429         92        768: 100%|██████████| 57/57 [00:18<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.82it/s]

                   all        181       1004      0.854      0.869      0.905      0.772



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      13/80      5.92G     0.4897     0.4197     0.9508         64        768: 100%|██████████| 57/57 [00:17<00:00,  3.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.52it/s]

                   all        181       1004      0.841      0.862       0.88      0.736



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      14/80      5.88G     0.4707     0.4029     0.9384         80        768: 100%|██████████| 57/57 [00:18<00:00,  3.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.82it/s]

                   all        181       1004      0.845      0.867      0.902      0.772



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      15/80      5.89G      0.457     0.3901     0.9418         56        768: 100%|██████████| 57/57 [00:17<00:00,  3.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.20it/s]

                   all        181       1004      0.845      0.871      0.889      0.777



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      16/80      5.92G     0.4492     0.3898     0.9313         68        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.75it/s]

                   all        181       1004      0.875      0.882      0.912      0.796



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      17/80      5.92G     0.4535     0.3908     0.9306         52        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.51it/s]

                   all        181       1004      0.837      0.898      0.895      0.771



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      18/80      5.91G     0.4458     0.3747     0.9283        105        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.50it/s]

                   all        181       1004      0.888      0.848      0.908      0.788



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      19/80      5.87G     0.4407     0.3686     0.9283         58        768: 100%|██████████| 57/57 [00:18<00:00,  3.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.55it/s]

                   all        181       1004      0.856      0.863      0.905      0.786



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      20/80      5.92G     0.4385      0.365     0.9233         55        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.60it/s]

                   all        181       1004      0.878      0.849       0.89      0.771



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      21/80      5.89G     0.4325     0.3622     0.9256         94        768: 100%|██████████| 57/57 [00:18<00:00,  3.03it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.12it/s]

                   all        181       1004      0.839      0.884      0.895      0.787



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      22/80       5.9G     0.4128     0.3452     0.9139         70        768: 100%|██████████| 57/57 [00:19<00:00,  2.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.79it/s]

                   all        181       1004      0.868      0.881      0.905      0.793



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      23/80       5.9G     0.4166     0.3467     0.9139         60        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.72it/s]

                   all        181       1004       0.85      0.858      0.898      0.769



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      24/80      5.93G     0.4101     0.3313     0.9168         82        768: 100%|██████████| 57/57 [00:18<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.83it/s]

                   all        181       1004      0.842      0.894      0.903      0.802



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      25/80      5.93G     0.4081     0.3369     0.9135         70        768: 100%|██████████| 57/57 [00:18<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.51it/s]

                   all        181       1004       0.87      0.852      0.904      0.783



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      26/80      5.93G     0.4048     0.3275     0.9125         71        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.75it/s]

                   all        181       1004      0.861      0.854      0.897      0.792



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      27/80      5.87G     0.4022     0.3258     0.9108         76        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.55it/s]

                   all        181       1004      0.864      0.868        0.9      0.786



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      28/80      5.93G     0.3971      0.317     0.9096         83        768: 100%|██████████| 57/57 [00:18<00:00,  3.06it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.72it/s]

                   all        181       1004      0.888      0.859      0.905      0.799



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      29/80      5.91G     0.3845      0.314     0.9012         54        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.70it/s]

                   all        181       1004      0.865       0.87      0.911      0.805



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      30/80      5.88G     0.3828     0.3109     0.9053         95        768: 100%|██████████| 57/57 [00:18<00:00,  3.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.61it/s]

                   all        181       1004      0.852      0.897      0.907      0.803



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      31/80       5.9G     0.3896     0.3052      0.907         90        768: 100%|██████████| 57/57 [00:18<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.78it/s]

                   all        181       1004      0.853      0.893      0.886      0.782



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      32/80       5.9G      0.384     0.3111     0.9082         79        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.41it/s]

                   all        181       1004      0.858      0.874      0.903      0.794



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      33/80      5.91G      0.378     0.3036     0.9015         82        768: 100%|██████████| 57/57 [00:18<00:00,  3.06it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.74it/s]

                   all        181       1004      0.864      0.894      0.909      0.802



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      34/80      5.88G     0.3715     0.2967     0.8974        132        768: 100%|██████████| 57/57 [00:18<00:00,  3.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.33it/s]

                   all        181       1004      0.862      0.894      0.909      0.806



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      35/80       5.9G     0.3647     0.2908     0.8951         59        768: 100%|██████████| 57/57 [00:18<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.76it/s]

                   all        181       1004      0.867      0.861      0.907      0.797



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      36/80      5.94G     0.3722        0.3     0.9063         97        768: 100%|██████████| 57/57 [00:18<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.32it/s]

                   all        181       1004      0.856      0.887      0.907      0.805



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      37/80      5.93G     0.3531     0.2832     0.8922        111        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.51it/s]

                   all        181       1004      0.862      0.897      0.915      0.815



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      38/80      5.89G      0.352      0.278     0.8894        103        768: 100%|██████████| 57/57 [00:18<00:00,  3.06it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.67it/s]

                   all        181       1004      0.863      0.901      0.904      0.805



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      39/80      5.88G      0.352     0.2826     0.8889         90        768: 100%|██████████| 57/57 [00:18<00:00,  3.06it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.53it/s]

                   all        181       1004      0.877      0.874       0.91      0.805



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      40/80      5.89G     0.3498     0.2797     0.8923         89        768: 100%|██████████| 57/57 [00:18<00:00,  3.01it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.58it/s]

                   all        181       1004      0.861      0.878      0.914      0.817



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      41/80      5.91G     0.3372     0.2727     0.8905         70        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.42it/s]

                   all        181       1004      0.875      0.876      0.918      0.813



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      42/80      5.88G     0.3376     0.2679     0.8852         86        768: 100%|██████████| 57/57 [00:18<00:00,  3.06it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.84it/s]

                   all        181       1004      0.851      0.896      0.905      0.808



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      43/80      5.88G     0.3442     0.2843     0.8856         76        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.74it/s]

                   all        181       1004      0.863      0.854      0.898      0.799



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      44/80      5.88G     0.3321     0.2653     0.8865         87        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.52it/s]

                   all        181       1004      0.854      0.901       0.92      0.819



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      45/80      5.87G      0.329     0.2593      0.884         67        768: 100%|██████████| 57/57 [00:18<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.79it/s]

                   all        181       1004      0.864      0.899      0.918      0.813



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      46/80      5.91G     0.3348     0.2702      0.888         57        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.39it/s]

                   all        181       1004      0.884      0.875       0.92      0.821



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      47/80      5.85G     0.3251     0.2593     0.8833         73        768: 100%|██████████| 57/57 [00:18<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.74it/s]

                   all        181       1004       0.88      0.876      0.917      0.809



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      48/80      5.91G     0.3223     0.2534     0.8794         98        768: 100%|██████████| 57/57 [00:18<00:00,  3.06it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.36it/s]

                   all        181       1004      0.891      0.864      0.913      0.816



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      49/80       5.9G     0.3205     0.2505     0.8815         78        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.70it/s]

                   all        181       1004      0.892      0.861      0.914      0.818



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      50/80       5.9G     0.3162     0.2518     0.8795         97        768: 100%|██████████| 57/57 [00:18<00:00,  3.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.73it/s]

                   all        181       1004      0.864       0.87      0.903      0.806



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      51/80      5.85G     0.3129      0.244     0.8799         69        768: 100%|██████████| 57/57 [00:18<00:00,  3.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.57it/s]

                   all        181       1004      0.884      0.857      0.914      0.818



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      52/80      5.93G     0.3087     0.2397     0.8803         81        768: 100%|██████████| 57/57 [00:18<00:00,  3.03it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.46it/s]

                   all        181       1004      0.879       0.87      0.913      0.816



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      53/80      5.85G     0.3006     0.2422     0.8732         94        768: 100%|██████████| 57/57 [00:19<00:00,  3.00it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.67it/s]

                   all        181       1004      0.877      0.884      0.925      0.828



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      54/80       5.9G     0.3053     0.2369      0.873         80        768: 100%|██████████| 57/57 [00:19<00:00,  2.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.48it/s]

                   all        181       1004      0.886      0.878      0.921      0.818



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      55/80      5.88G     0.2955     0.2307     0.8723         73        768: 100%|██████████| 57/57 [00:18<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.75it/s]

                   all        181       1004      0.872       0.88      0.913      0.825



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      56/80      5.89G     0.3005     0.2337      0.876         91        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.49it/s]

                   all        181       1004      0.872      0.881      0.921      0.819



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      57/80      5.93G     0.2933     0.2304     0.8734         98        768: 100%|██████████| 57/57 [00:18<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.68it/s]

                   all        181       1004      0.891      0.888      0.923      0.824



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      58/80      5.91G     0.2921     0.2286      0.871         73        768: 100%|██████████| 57/57 [00:18<00:00,  3.03it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.32it/s]

                   all        181       1004      0.855      0.885      0.911      0.816



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      59/80       5.9G      0.293     0.2274     0.8747         72        768: 100%|██████████| 57/57 [00:18<00:00,  3.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.62it/s]

                   all        181       1004      0.853      0.898      0.918      0.823



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      60/80      5.89G     0.2845     0.2241     0.8673         76        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.51it/s]

                   all        181       1004      0.863      0.901      0.922      0.828



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      61/80      5.92G     0.2822      0.221     0.8678         54        768: 100%|██████████| 57/57 [00:18<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.72it/s]

                   all        181       1004      0.888       0.88      0.919      0.825



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      62/80      5.89G     0.2779     0.2177     0.8702         96        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.80it/s]

                   all        181       1004      0.872      0.897      0.915      0.821



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      63/80      5.78G     0.2776     0.2174     0.8649         82        768: 100%|██████████| 57/57 [00:18<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.31it/s]

                   all        181       1004       0.87      0.894      0.921      0.827



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      64/80      5.93G     0.2716     0.2156      0.863         81        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.37it/s]

                   all        181       1004      0.861      0.897      0.924      0.831



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      65/80      5.91G     0.2669      0.209     0.8633         57        768: 100%|██████████| 57/57 [00:18<00:00,  3.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.48it/s]

                   all        181       1004      0.875      0.896      0.922      0.828



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      66/80      5.92G     0.2649     0.2087     0.8578         49        768: 100%|██████████| 57/57 [00:18<00:00,  3.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.71it/s]

                   all        181       1004      0.874      0.849      0.911      0.818



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      67/80      5.89G     0.2662     0.2063      0.862        117        768: 100%|██████████| 57/57 [00:18<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.23it/s]

                   all        181       1004      0.872      0.894      0.919      0.827



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      68/80       5.9G     0.2626     0.2069     0.8606         88        768: 100%|██████████| 57/57 [00:18<00:00,  3.06it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.49it/s]

                   all        181       1004      0.869      0.884       0.92      0.828



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      69/80      5.93G      0.258     0.1988     0.8602         70        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.66it/s]

                   all        181       1004      0.867      0.876       0.92      0.827



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      70/80      5.93G     0.2574     0.2007      0.862         72        768: 100%|██████████| 57/57 [00:18<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.57it/s]

                   all        181       1004      0.874      0.891      0.922       0.83


Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      71/80      5.86G     0.4089      0.319      0.923         46        768: 100%|██████████| 57/57 [00:19<00:00,  2.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.57it/s]

                   all        181       1004      0.879       0.88      0.917      0.822



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      72/80      5.91G     0.4024     0.2893      0.901         36        768: 100%|██████████| 57/57 [00:18<00:00,  3.03it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.85it/s]

                   all        181       1004       0.86      0.914      0.915      0.818



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      73/80      5.88G     0.3849     0.2764     0.8946         52        768: 100%|██████████| 57/57 [00:18<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.70it/s]

                   all        181       1004      0.856      0.924      0.921      0.828



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      74/80      5.89G     0.3853     0.2633     0.8971         42        768: 100%|██████████| 57/57 [00:18<00:00,  3.03it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.85it/s]

                   all        181       1004      0.859      0.911      0.919      0.819



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      75/80      5.86G     0.3844     0.2578     0.8977         40        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.54it/s]

                   all        181       1004      0.877      0.891      0.919      0.823



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      76/80      5.93G     0.3729     0.2473       0.89         47        768: 100%|██████████| 57/57 [00:18<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.46it/s]

                   all        181       1004      0.872      0.899      0.923      0.829



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      77/80      5.88G     0.3709      0.248     0.8864         42        768: 100%|██████████| 57/57 [00:18<00:00,  3.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.76it/s]

                   all        181       1004      0.857      0.917      0.921      0.827



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      78/80      5.93G     0.3631     0.2358     0.8843         50        768: 100%|██████████| 57/57 [00:18<00:00,  3.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.69it/s]

                   all        181       1004      0.861      0.919       0.92      0.823



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      79/80      5.86G     0.3601     0.2306     0.8851         40        768: 100%|██████████| 57/57 [00:18<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.52it/s]

                   all        181       1004      0.872      0.902      0.919      0.825



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      80/80      5.91G     0.3556      0.231     0.8774         58        768: 100%|██████████| 57/57 [00:18<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  3.49it/s]

                   all        181       1004      0.856      0.922       0.92      0.825



80 epochs completed in 0.527 hours.
Optimizer stripped from runs\detect\compare_yolov8s_dishdet\weights\last.pt, 20.0MB
Optimizer stripped from runs\detect\compare_yolov8s_dishdet\weights\best.pt, 20.0MB

Validating runs\detect\compare_yolov8s_dishdet\weights\best.pt...
Ultralytics 8.3.5  Python-3.11.9 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)
Model summary (fused): 186 layers, 9,828,051 parameters, 0 gradients


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.34it/s]


                   all        181       1004      0.861      0.898      0.923      0.831
Speed: 0.2ms preprocess, 5.4ms inference, 0.0ms loss, 1.6ms postprocess per image
Results saved to runs\detect\compare_yolov8s_dishdet

[YOLOv8-s] Best weights: runs\detect\compare_yolov8s_dishdet\weights\best.pt

Validating best YOLOv8-s weights on validation set
Ultralytics 8.3.5  Python-3.11.9 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)
Model summary (fused): 186 layers, 9,828,051 parameters, 0 gradients


val: Scanning C:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\data_yolo_dish\valid\labels.cache... 181 images, 0 backgrounds, 0 corrupt: 100%|██████████| 181/181 [00:00<?, ?it/s]

WARNING  Box and segment counts should be equal, but got len(segments) = 28, len(boxes) = 1004. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 12/12 [00:05<00:00,  2.36it/s]


                   all        181       1004      0.864      0.896      0.923      0.831
Speed: 0.5ms preprocess, 8.7ms inference, 0.0ms loss, 2.4ms postprocess per image
Results saved to c:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\runs\detect\val7
[YOLOv8-s] P=0.864, R=0.896, mAP50=0.923, mAP50-95=0.831


## Train YOLO11-n (Latest Architecture)

Train YOLO11-n, the newest YOLO architecture released in late 2024. This model features improved accuracy through architectural enhancements while maintaining a small parameter count.

In [11]:
print("=" * 70)
print("Training YOLO11-n for dish detection")
print("=" * 70)

model_y11n = YOLO("yolo11n.pt")

result_y11n = model_y11n.train(
    data=DATA_YAML,
    epochs=EPOCHS,
    imgsz=IMG_SIZE,
    batch=BATCH_SIZE,
    device=DEVICE,
    name="compare_yolo11n_dishdet",
    project="runs/detect",
    cache="ram",
    seed=0,
    deterministic=True,
)

BEST_YOLO11N = os.path.join(
    "runs", "detect", "compare_yolo11n_dishdet", "weights", "best.pt"
)
print(f"\n[YOLO11-n] Best weights: {BEST_YOLO11N}")

print("\nValidating best YOLO11-n weights on validation set")
y11n_best_model = YOLO(BEST_YOLO11N)
metrics_y11n = y11n_best_model.val(
    data=DATA_YAML,
    split="val",
    imgsz=IMG_SIZE,
    batch=BATCH_SIZE,
)

print(
    f"[YOLO11-n] P={metrics_y11n.box.mp:.3f}, "
    f"R={metrics_y11n.box.mr:.3f}, "
    f"mAP50={metrics_y11n.box.map50:.3f}, "
    f"mAP50-95={metrics_y11n.box.map:.3f}"
)


Training YOLO11-n for dish detection
New https://pypi.org/project/ultralytics/8.3.233 available  Update with 'pip install -U ultralytics'
Ultralytics 8.3.5  Python-3.11.9 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)
engine\trainer: task=detect, mode=train, model=yolo11n.pt, data=data_yolo_dish/data.yaml, epochs=80, time=None, patience=100, batch=16, imgsz=768, save=True, save_period=-1, cache=ram, device=0, workers=8, project=runs/detect, name=compare_yolo11n_dishdet, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=

train: Scanning C:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\data_yolo_dish\train\labels.cache... 903 images, 0 backgrounds, 0 corrupt: 100%|██████████| 903/903 [00:00<?, ?it/s]

WARNING  Box and segment counts should be equal, but got len(segments) = 83, len(boxes) = 5212. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.
train: 1.5GB RAM required to cache images with 50% safety margin but only 1.2/31.7GB available, not caching images 



val: Scanning C:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\data_yolo_dish\valid\labels.cache... 181 images, 0 backgrounds, 0 corrupt: 100%|██████████| 181/181 [00:00<?, ?it/s]

WARNING  Box and segment counts should be equal, but got len(segments) = 28, len(boxes) = 1004. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.
WARNING  cache='ram' may produce non-deterministic training results. Consider cache='disk' as a deterministic alternative if your disk space allows.



val: Caching images (0.2GB RAM): 100%|██████████| 181/181 [00:00<00:00, 857.76it/s]


Plotting labels to runs\detect\compare_yolo11n_dishdet\labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 81 weight(decay=0.0), 88 weight(decay=0.0005), 87 bias(decay=0.0)
TensorBoard: model graph visualization added 
Image sizes 768 train, 768 val
Using 8 dataloader workers
Logging results to runs\detect\compare_yolo11n_dishdet
Starting training for 80 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/80      3.73G     0.6832      1.618      1.069        105        768: 100%|██████████| 57/57 [00:12<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.94it/s]

                   all        181       1004      0.843      0.837      0.873       0.73



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       2/80      4.03G     0.6616     0.8429      1.049         98        768: 100%|██████████| 57/57 [00:11<00:00,  4.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.87it/s]

                   all        181       1004      0.798      0.845      0.877      0.714



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       3/80      3.99G     0.6806     0.7568      1.043         83        768: 100%|██████████| 57/57 [00:11<00:00,  4.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.12it/s]

                   all        181       1004      0.828      0.861      0.879      0.717



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       4/80         4G     0.6683     0.7267      1.047         80        768: 100%|██████████| 57/57 [00:11<00:00,  4.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.64it/s]

                   all        181       1004      0.809      0.706      0.799      0.649



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       5/80      3.99G     0.6491     0.6888      1.026         54        768: 100%|██████████| 57/57 [00:11<00:00,  4.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.84it/s]

                   all        181       1004      0.801      0.823       0.86      0.671



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       6/80      3.99G     0.6249     0.6467      1.009         83        768: 100%|██████████| 57/57 [00:11<00:00,  4.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.20it/s]

                   all        181       1004      0.806      0.892      0.869      0.742



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       7/80      4.04G     0.6289     0.6296      1.012         84        768: 100%|██████████| 57/57 [00:11<00:00,  4.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.39it/s]

                   all        181       1004      0.855      0.878      0.899      0.767



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       8/80         4G     0.6191     0.6002      1.002         82        768: 100%|██████████| 57/57 [00:11<00:00,  4.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.31it/s]

                   all        181       1004      0.847       0.88      0.914      0.774



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       9/80      4.02G     0.6178     0.5915      1.007         95        768: 100%|██████████| 57/57 [00:11<00:00,  4.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.73it/s]

                   all        181       1004      0.856      0.893      0.906      0.781



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      10/80      3.99G     0.6012     0.5693     0.9995        140        768: 100%|██████████| 57/57 [00:11<00:00,  4.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.27it/s]

                   all        181       1004      0.866      0.874      0.916      0.799



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      11/80      3.99G     0.6122     0.5641       1.01        110        768: 100%|██████████| 57/57 [00:11<00:00,  4.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.31it/s]

                   all        181       1004      0.864       0.89       0.91      0.782



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      12/80      3.93G     0.5989     0.5516     0.9964         78        768: 100%|██████████| 57/57 [00:11<00:00,  4.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.48it/s]

                   all        181       1004       0.87      0.882      0.916      0.793



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      13/80      4.03G     0.5905     0.5288     0.9985         82        768: 100%|██████████| 57/57 [00:11<00:00,  4.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.31it/s]

                   all        181       1004      0.812      0.835      0.868      0.752



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      14/80      4.03G     0.5839      0.541     0.9916         75        768: 100%|██████████| 57/57 [00:11<00:00,  4.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.13it/s]

                   all        181       1004       0.86      0.877      0.916       0.79



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      15/80      4.03G     0.5834     0.5365      0.993         60        768: 100%|██████████| 57/57 [00:11<00:00,  4.81it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.28it/s]

                   all        181       1004      0.863       0.88      0.922       0.79



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      16/80      4.01G     0.5795     0.5308     0.9866         67        768: 100%|██████████| 57/57 [00:11<00:00,  4.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.32it/s]

                   all        181       1004      0.865      0.869       0.91      0.791



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      17/80      4.05G     0.5734     0.5302     0.9856         53        768: 100%|██████████| 57/57 [00:11<00:00,  4.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.15it/s]

                   all        181       1004      0.851      0.911      0.915      0.802



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      18/80      4.03G     0.5664     0.5007     0.9782        114        768: 100%|██████████| 57/57 [00:11<00:00,  4.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.94it/s]

                   all        181       1004      0.855      0.889      0.916      0.783



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      19/80      3.98G     0.5648     0.5074     0.9782         75        768: 100%|██████████| 57/57 [00:11<00:00,  4.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.11it/s]

                   all        181       1004      0.854      0.842        0.9      0.772



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      20/80         4G     0.5673     0.4959     0.9849         69        768: 100%|██████████| 57/57 [00:11<00:00,  4.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.91it/s]

                   all        181       1004      0.849      0.903      0.911      0.791



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      21/80      4.05G     0.5779     0.5087     0.9904         83        768: 100%|██████████| 57/57 [00:11<00:00,  4.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.53it/s]

                   all        181       1004      0.852      0.883      0.918      0.801



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      22/80      4.01G     0.5491     0.4926     0.9759         80        768: 100%|██████████| 57/57 [00:11<00:00,  4.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.28it/s]

                   all        181       1004      0.856      0.892      0.907      0.799



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      23/80      4.05G     0.5478     0.4864     0.9761         63        768: 100%|██████████| 57/57 [00:11<00:00,  4.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.24it/s]

                   all        181       1004      0.889       0.88      0.914      0.796



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      24/80      4.03G     0.5509     0.4693      0.972         76        768: 100%|██████████| 57/57 [00:11<00:00,  4.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.47it/s]

                   all        181       1004      0.869      0.885       0.92      0.814



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      25/80      4.03G     0.5389     0.4758     0.9729         74        768: 100%|██████████| 57/57 [00:11<00:00,  4.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.40it/s]

                   all        181       1004      0.866      0.897      0.928      0.816



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      26/80      4.03G     0.5445      0.467     0.9713         79        768: 100%|██████████| 57/57 [00:11<00:00,  4.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.92it/s]

                   all        181       1004      0.878      0.871      0.925      0.815



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      27/80      4.01G     0.5354     0.4748     0.9657         83        768: 100%|██████████| 57/57 [00:11<00:00,  5.00it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.32it/s]

                   all        181       1004      0.876        0.9      0.918      0.806



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      28/80      4.02G     0.5468     0.4649     0.9712         63        768: 100%|██████████| 57/57 [00:11<00:00,  4.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.89it/s]

                   all        181       1004      0.882      0.882      0.924      0.805



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      29/80      4.02G     0.5293     0.4581     0.9639         57        768: 100%|██████████| 57/57 [00:11<00:00,  4.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.40it/s]

                   all        181       1004       0.86      0.908      0.928      0.814



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      30/80      4.03G      0.536     0.4595     0.9689         76        768: 100%|██████████| 57/57 [00:11<00:00,  4.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.49it/s]

                   all        181       1004      0.872      0.899      0.925      0.816



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      31/80      4.03G      0.524      0.449     0.9599         96        768: 100%|██████████| 57/57 [00:11<00:00,  4.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.91it/s]

                   all        181       1004      0.878      0.902      0.927      0.825



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      32/80      4.02G     0.5232      0.444     0.9649         95        768: 100%|██████████| 57/57 [00:11<00:00,  4.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.55it/s]

                   all        181       1004       0.87       0.88      0.914      0.808



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      33/80      4.02G      0.535     0.4632      0.961         75        768: 100%|██████████| 57/57 [00:11<00:00,  4.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.24it/s]

                   all        181       1004      0.871      0.898      0.923      0.813



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      34/80      4.03G     0.5232     0.4439     0.9597        131        768: 100%|██████████| 57/57 [00:11<00:00,  4.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.05it/s]

                   all        181       1004      0.869      0.895       0.92      0.813



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      35/80         4G     0.5241     0.4431     0.9567         60        768: 100%|██████████| 57/57 [00:11<00:00,  4.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.52it/s]

                   all        181       1004      0.885       0.88      0.924       0.82



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      36/80      4.03G     0.5262     0.4506      0.965         98        768: 100%|██████████| 57/57 [00:11<00:00,  4.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.37it/s]

                   all        181       1004      0.832      0.887      0.908      0.798



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      37/80      3.99G     0.5109     0.4219     0.9534         83        768: 100%|██████████| 57/57 [00:11<00:00,  4.80it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.75it/s]

                   all        181       1004      0.871      0.912      0.928       0.83



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      38/80      3.99G     0.5035     0.4183     0.9469        114        768: 100%|██████████| 57/57 [00:11<00:00,  4.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.15it/s]


                   all        181       1004       0.86      0.911       0.92      0.811

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      39/80      4.19G     0.5079      0.427     0.9485         84        768: 100%|██████████| 57/57 [00:11<00:00,  4.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.83it/s]

                   all        181       1004      0.864      0.906      0.927      0.826



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      40/80      4.04G     0.5072     0.4193      0.954         90        768: 100%|██████████| 57/57 [00:11<00:00,  4.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.46it/s]

                   all        181       1004      0.861      0.881      0.918      0.816



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      41/80      4.03G     0.5102     0.4146     0.9526         72        768: 100%|██████████| 57/57 [00:11<00:00,  4.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.03it/s]

                   all        181       1004       0.88       0.88      0.918      0.807



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      42/80      4.01G     0.5066     0.4131     0.9497         86        768: 100%|██████████| 57/57 [00:11<00:00,  4.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.46it/s]

                   all        181       1004      0.857      0.863      0.914      0.813



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      43/80      4.03G     0.5077     0.4154     0.9552         72        768: 100%|██████████| 57/57 [00:11<00:00,  4.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.44it/s]

                   all        181       1004      0.851      0.915      0.928      0.837



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      44/80      3.99G     0.5069     0.4121     0.9509         92        768: 100%|██████████| 57/57 [00:11<00:00,  4.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.12it/s]

                   all        181       1004      0.862      0.878      0.916      0.825



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      45/80      3.99G     0.5048     0.4119     0.9505         73        768: 100%|██████████| 57/57 [00:11<00:00,  4.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.18it/s]

                   all        181       1004      0.859      0.897      0.923      0.827



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      46/80      4.03G     0.5019     0.4041     0.9489         51        768: 100%|██████████| 57/57 [00:11<00:00,  4.75it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.62it/s]

                   all        181       1004      0.863       0.89      0.931      0.833



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      47/80      4.01G     0.4884     0.3921     0.9446         81        768: 100%|██████████| 57/57 [00:11<00:00,  4.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.22it/s]

                   all        181       1004      0.855      0.903      0.924      0.819



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      48/80         4G      0.497     0.4016     0.9459         94        768: 100%|██████████| 57/57 [00:11<00:00,  4.79it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.47it/s]

                   all        181       1004      0.881      0.904      0.927      0.834



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      49/80      4.01G      0.486      0.394     0.9408         80        768: 100%|██████████| 57/57 [00:11<00:00,  4.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.30it/s]

                   all        181       1004      0.879      0.899      0.925      0.834



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      50/80      3.99G     0.4985     0.4025      0.948        113        768: 100%|██████████| 57/57 [00:11<00:00,  4.81it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.23it/s]

                   all        181       1004      0.834      0.893       0.91      0.811



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      51/80      4.05G     0.4775     0.3809     0.9366         76        768: 100%|██████████| 57/57 [00:11<00:00,  4.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.37it/s]

                   all        181       1004      0.884      0.879      0.927      0.836



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      52/80      3.98G     0.4834     0.3844     0.9403         75        768: 100%|██████████| 57/57 [00:11<00:00,  4.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.10it/s]

                   all        181       1004      0.892      0.878      0.931      0.836



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      53/80       3.8G     0.4767     0.3789     0.9283         88        768: 100%|██████████| 57/57 [00:11<00:00,  4.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.51it/s]

                   all        181       1004      0.874      0.886      0.929      0.833



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      54/80      3.98G     0.4816     0.3781     0.9376         89        768: 100%|██████████| 57/57 [00:11<00:00,  4.80it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.11it/s]

                   all        181       1004       0.87      0.897      0.929      0.834



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      55/80      4.04G     0.4741     0.3669     0.9319         79        768: 100%|██████████| 57/57 [00:11<00:00,  4.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.75it/s]

                   all        181       1004      0.908       0.88      0.933      0.843



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      56/80      4.03G      0.482     0.3625     0.9357         87        768: 100%|██████████| 57/57 [00:11<00:00,  4.80it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.32it/s]

                   all        181       1004      0.858      0.921      0.931      0.839



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      57/80      4.01G      0.476     0.3647     0.9339        113        768: 100%|██████████| 57/57 [00:11<00:00,  4.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.97it/s]

                   all        181       1004       0.85       0.92      0.924      0.838



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      58/80      4.02G     0.4715     0.3642     0.9329         99        768: 100%|██████████| 57/57 [00:11<00:00,  4.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.46it/s]

                   all        181       1004       0.83      0.912      0.921      0.833



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      59/80         4G     0.4686     0.3582     0.9325         62        768: 100%|██████████| 57/57 [00:11<00:00,  4.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.93it/s]

                   all        181       1004      0.872        0.9      0.933      0.841



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      60/80      4.03G     0.4605     0.3473     0.9237         89        768: 100%|██████████| 57/57 [00:11<00:00,  4.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.45it/s]

                   all        181       1004      0.892      0.868      0.922       0.83



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      61/80      3.99G     0.4613     0.3532     0.9256         61        768: 100%|██████████| 57/57 [00:11<00:00,  4.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.56it/s]

                   all        181       1004       0.88      0.875      0.923      0.831



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      62/80      4.02G     0.4603     0.3405     0.9286         95        768: 100%|██████████| 57/57 [00:11<00:00,  4.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.49it/s]

                   all        181       1004      0.862      0.901      0.925      0.838



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      63/80      4.02G     0.4608     0.3492     0.9237         83        768: 100%|██████████| 57/57 [00:11<00:00,  4.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.47it/s]

                   all        181       1004      0.857      0.885      0.921      0.837



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      64/80      4.04G     0.4646     0.3475     0.9255         68        768: 100%|██████████| 57/57 [00:11<00:00,  4.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.47it/s]

                   all        181       1004      0.859      0.893      0.921      0.835



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      65/80      4.03G     0.4555     0.3331      0.925         65        768: 100%|██████████| 57/57 [00:11<00:00,  4.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.59it/s]

                   all        181       1004      0.882      0.901       0.93      0.844



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      66/80      4.05G     0.4619     0.3487     0.9281         36        768: 100%|██████████| 57/57 [00:11<00:00,  4.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.10it/s]

                   all        181       1004       0.87       0.89      0.927       0.84



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      67/80      4.05G     0.4503     0.3335     0.9198        105        768: 100%|██████████| 57/57 [00:11<00:00,  4.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.16it/s]

                   all        181       1004      0.879      0.894      0.929      0.842



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      68/80      3.99G     0.4523     0.3382     0.9259         84        768: 100%|██████████| 57/57 [00:11<00:00,  4.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.28it/s]

                   all        181       1004        0.9      0.877      0.929      0.842



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      69/80       3.8G     0.4504     0.3258     0.9202         70        768: 100%|██████████| 57/57 [00:11<00:00,  4.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.43it/s]

                   all        181       1004      0.887      0.895      0.929      0.845



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      70/80      3.93G      0.455     0.3296     0.9281         72        768: 100%|██████████| 57/57 [00:11<00:00,  4.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.27it/s]

                   all        181       1004      0.849      0.924       0.93      0.843


Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      71/80      4.24G      0.407     0.3056     0.8953         46        768: 100%|██████████| 57/57 [00:12<00:00,  4.65it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.24it/s]

                   all        181       1004      0.856      0.912      0.919      0.828



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      72/80         4G     0.4008     0.2769     0.8824         36        768: 100%|██████████| 57/57 [00:11<00:00,  5.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.16it/s]

                   all        181       1004      0.867      0.884       0.91      0.821



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      73/80      3.98G     0.3885     0.2641     0.8762         52        768: 100%|██████████| 57/57 [00:11<00:00,  4.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.60it/s]

                   all        181       1004      0.876      0.898      0.926       0.84



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      74/80      4.01G     0.3879     0.2588     0.8768         42        768: 100%|██████████| 57/57 [00:11<00:00,  4.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.09it/s]

                   all        181       1004      0.884      0.887      0.925      0.838



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      75/80      3.99G     0.3896     0.2566     0.8753         40        768: 100%|██████████| 57/57 [00:11<00:00,  5.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.92it/s]


                   all        181       1004      0.887      0.876      0.924      0.841

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      76/80      4.02G     0.3818     0.2495     0.8772         47        768: 100%|██████████| 57/57 [00:11<00:00,  4.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.15it/s]

                   all        181       1004      0.888      0.877      0.925      0.843



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      77/80      3.98G     0.3811     0.2535     0.8739         42        768: 100%|██████████| 57/57 [00:11<00:00,  5.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.30it/s]

                   all        181       1004      0.908      0.855      0.925      0.843



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      78/80      4.03G      0.375     0.2434     0.8725         50        768: 100%|██████████| 57/57 [00:11<00:00,  5.03it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.77it/s]

                   all        181       1004      0.896      0.878      0.927      0.841



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      79/80      3.98G      0.378     0.2424     0.8761         40        768: 100%|██████████| 57/57 [00:11<00:00,  4.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  5.00it/s]

                   all        181       1004      0.874      0.895      0.926      0.843



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      80/80      4.01G     0.3705     0.2406     0.8665         58        768: 100%|██████████| 57/57 [00:11<00:00,  4.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:01<00:00,  4.73it/s]

                   all        181       1004      0.883      0.892      0.929      0.847



80 epochs completed in 0.320 hours.
Optimizer stripped from runs\detect\compare_yolo11n_dishdet\weights\last.pt, 5.5MB
Optimizer stripped from runs\detect\compare_yolo11n_dishdet\weights\best.pt, 5.5MB

Validating runs\detect\compare_yolo11n_dishdet\weights\best.pt...
Ultralytics 8.3.5  Python-3.11.9 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)
YOLO11n summary (fused): 238 layers, 2,582,347 parameters, 0 gradients


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.64it/s]


                   all        181       1004      0.884      0.892      0.929      0.847
Speed: 0.2ms preprocess, 2.2ms inference, 0.0ms loss, 1.9ms postprocess per image
Results saved to runs\detect\compare_yolo11n_dishdet

[YOLO11-n] Best weights: runs\detect\compare_yolo11n_dishdet\weights\best.pt

Validating best YOLO11-n weights on validation set
Ultralytics 8.3.5  Python-3.11.9 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)
YOLO11n summary (fused): 238 layers, 2,582,347 parameters, 0 gradients


val: Scanning C:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\data_yolo_dish\valid\labels.cache... 181 images, 0 backgrounds, 0 corrupt: 100%|██████████| 181/181 [00:00<?, ?it/s]

WARNING  Box and segment counts should be equal, but got len(segments) = 28, len(boxes) = 1004. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 12/12 [00:03<00:00,  3.22it/s]


                   all        181       1004      0.884      0.892      0.928      0.847
Speed: 0.6ms preprocess, 4.8ms inference, 0.0ms loss, 0.9ms postprocess per image
Results saved to c:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\runs\detect\val8
[YOLO11-n] P=0.884, R=0.892, mAP50=0.928, mAP50-95=0.847


## Model Comparison

Evaluate all three models on the validation set and compare their performance metrics:
- **Precision (P)**: How many detected dishes are correct
- **Recall (R)**: How many actual dishes are detected
- **mAP50**: Mean Average Precision at IoU threshold 0.5
- **mAP50-95**: Mean Average Precision averaged across IoU thresholds 0.5-0.95

In [12]:
print("=" * 70)
print("Comparing YOLOv8-n vs YOLOv8-s vs YOLO11-n")
print("=" * 70)

# YOLOv8-n
model_y8n = YOLO(BEST_YOLOV8N)
metrics_y8n = model_y8n.val(
    data=DATA_YAML,
    split="val",
    imgsz=IMG_SIZE,
    batch=BATCH_SIZE,
    verbose=False,
)

# YOLOv8-s
metrics_y8s = y8s_best_model.val(
    data=DATA_YAML,
    split="val",
    imgsz=IMG_SIZE,
    batch=BATCH_SIZE,
    verbose=False,
)

# YOLO11-n
metrics_y11n = y11n_best_model.val(
    data=DATA_YAML,
    split="val",
    imgsz=IMG_SIZE,
    batch=BATCH_SIZE,
    verbose=False,
)

rows = [
    {
        "Model": "YOLOv8-n",
        "Params": "~3.2M (tiny)",
        "P": metrics_y8n.box.mp,
        "R": metrics_y8n.box.mr,
        "mAP50": metrics_y8n.box.map50,
        "mAP50-95": metrics_y8n.box.map,
    },
    {
        "Model": "YOLOv8-s",
        "Params": "~9.8M (small)",
        "P": metrics_y8s.box.mp,
        "R": metrics_y8s.box.mr,
        "mAP50": metrics_y8s.box.map50,
        "mAP50-95": metrics_y8s.box.map,
    },
    {
        "Model": "YOLO11-n",
        "Params": "~[n-size] (nano)",
        "P": metrics_y11n.box.mp,
        "R": metrics_y11n.box.mr,
        "mAP50": metrics_y11n.box.map50,
        "mAP50-95": metrics_y11n.box.map,
    },
]

df_compare = pd.DataFrame(rows)
df_compare


Comparing YOLOv8-n vs YOLOv8-s vs YOLO11-n
Ultralytics 8.3.5  Python-3.11.9 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)
Model summary (fused): 186 layers, 2,684,563 parameters, 0 gradients


val: Scanning C:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\data_yolo_dish\valid\labels.cache... 181 images, 0 backgrounds, 0 corrupt: 100%|██████████| 181/181 [00:00<?, ?it/s]

WARNING  Box and segment counts should be equal, but got len(segments) = 28, len(boxes) = 1004. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 12/12 [00:04<00:00,  2.94it/s]


                   all        181       1004      0.852      0.905      0.917      0.828
Speed: 0.7ms preprocess, 4.5ms inference, 0.0ms loss, 1.9ms postprocess per image
Results saved to c:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\runs\detect\val9
Ultralytics 8.3.5  Python-3.11.9 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)


val: Scanning C:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\data_yolo_dish\valid\labels.cache... 181 images, 0 backgrounds, 0 corrupt: 100%|██████████| 181/181 [00:00<?, ?it/s]

WARNING  Box and segment counts should be equal, but got len(segments) = 28, len(boxes) = 1004. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 12/12 [00:05<00:00,  2.27it/s]


                   all        181       1004      0.864      0.896      0.923      0.831
Speed: 0.5ms preprocess, 8.8ms inference, 0.0ms loss, 2.4ms postprocess per image
Results saved to c:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\runs\detect\val10
Ultralytics 8.3.5  Python-3.11.9 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)


val: Scanning C:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\data_yolo_dish\valid\labels.cache... 181 images, 0 backgrounds, 0 corrupt: 100%|██████████| 181/181 [00:00<?, ?it/s]

WARNING  Box and segment counts should be equal, but got len(segments) = 28, len(boxes) = 1004. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 12/12 [00:04<00:00,  2.83it/s]


                   all        181       1004      0.884      0.892      0.928      0.847
Speed: 0.5ms preprocess, 5.5ms inference, 0.0ms loss, 1.7ms postprocess per image
Results saved to c:\Users\Jun Wei\Desktop\SIT\SIT Y2T1\Computer Vision and Deep Learning\Proj\food_classifier\runs\detect\val11


,Model,Params,P,R,mAP50,mAP50-95
0,YOLOv8-n,~3.2M (tiny),0.852026,0.905378,0.917199,0.827593
1,YOLOv8-s,~9.8M (small),0.864029,0.896414,0.922875,0.830966
2,YOLO11-n,~[n-size] (nano),0.883558,0.891817,0.928259,0.846787


## Results & Conclusion

### Performance Summary

| Model | Params | P | R | mAP50 | mAP50-95 | Inference |
|-------|--------|---|---|-------|----------|-----------|
| YOLOv8-n | ~3.2M | 0.852 | 0.905 | 0.917 | 0.828 | 4.5ms |
| YOLOv8-s | ~9.8M | 0.864 | 0.896 | 0.923 | 0.831 | 8.8ms |
| YOLO11-n | ~2.6M | 0.884 | 0.892 | 0.928 | 0.847 | 5.5ms |

### Why We Choose YOLOv8-n

Although YOLO11-n achieves slightly higher accuracy metrics, we select **YOLOv8-n** for deployment based on:

1. **Fastest inference (4.5ms)** - Critical for real-time webcam streaming in our Gradio app
2. **Highest recall (0.905)** - Detects more dishes with fewer misses, important for food classification pipeline
3. **Production stability** - YOLOv8 is more mature with better documentation and community support
4. **Marginal accuracy difference** - Only ~1-2% behind YOLO11-n, negligible for our use case

For applications requiring maximum accuracy over speed, YOLO11-n would be the better choice.